src/manifolds/SymplecticStiefel.jl

```diff
@@ -0,0 +1,554 @@
+
+    if !isapprox(hamiltonian_identity_norm, 0; kwargs...)
+        return DomainError(
+            hamiltonian_identity_norm,
+            (
+                "The matrix X is not in the tangent space at point p of the" *
+                " $(M) manifold, as p^{+}X is not a Hamiltonian matrix."
+            ),
+        )
+    end
+    return nothing
+end
+
+@doc raw"""
+    inner(M::SymplecticStiefel{n, k}, p, X. Y)
+
+Compute the Riemannian inner product ``g^{\operatorname{SpSt}}`` at
+``p \in \operatorname{SpSt}`` between tangent vectors ``X, X \in T_p\operatorname{SpSt}``.
+Given by Proposition 3.10 in [^Bendokat2021].
+````math
+g^{\operatorname{SpSt}}_p(X, Y)
+    = \operatorname{tr}\left(X^T\left(I_{2n} -
+        \frac{1}{2}Q_{2n}^Tp(p^Tp)^{-1}p^TQ_{2n}\right)Y(p^Tp)^{-1}\right).
+````
+"""
+function inner(::SymplecticStiefel{n,k}, p, X, Y) where {n,k}
+    Q = SymplecticMatrix(p, X, Y)
+    # Procompute lu(p'p) since we solve a^{-1}* 3 times
+    a = lu(p' * p) # note that p'p is symmetric, thus so is its inverse c=a^{-1}
+    b = Q' * p
+    # we split the original trace into two one with I->(X'Yc)
+    # and the other with 1/2 X'b c b' Y c
+    # a) we permute X' and Y c to c^TY^TX = a\(Y'X) (avoids a large interims matrix)
+    # b) we permute Y c up front, the center term is symmetric, so we get cY'b c b' X
+    # and (b'X) again avoids a large interims matrix, so does Y'b.
+    return tr(a \ (Y' * X)) - (1 / 2) * tr(a \ ((Y' * b) * (a \ (b' * X))))
+end
+
+@doc raw"""
+    inv(::SymplecticStiefel{n, k}, A)
+    inv!(::SymplecticStiefel{n, k}, q, p)
+
+Compute the symplectic inverse ``A^+`` of matrix ``A ∈ ℝ^{2n × 2k}``. Given a matrix
+````math
+A ∈ ℝ^{2n × 2k},\quad
+A =
+\begin{bmatrix}
+A_{1, 1} & A_{1, 2} \\
+A_{2, 1} & A_{2, 2}
+\end{bmatrix},\; A_{i, j} \in ℝ^{2n × 2k}
+````
+the symplectic inverse is defined as:
+````math
+A^{+} := Q_{2k}^T A^T Q_{2n},
+````
+where
+````math
+Q_{2n} =
+\begin{bmatrix}
+0_n & I_n \\
+ -I_n & 0_n
+\end{bmatrix}.
+````
+For any ``p \in \operatorname{SpSt}(2n, 2k)`` we have that ``p^{+}p = I_{2k}``.
+
+The symplectic inverse of a matrix A can be expressed explicitly as:
+````math
+A^{+} =
+\begin{bmatrix}
+  A_{2, 2}^T & -A_{1, 2}^T \\[1.2mm]
+ -A_{2, 1}^T &  A_{1, 1}^T
+\end{bmatrix}.
+````
+"""
+function Base.inv(M::SymplecticStiefel{n,k}, p) where {n,k}
+    q = similar(p')
+    return inv!(M, q, p)
+end
+
+function inv!(::SymplecticStiefel{n,k}, q, p) where {n,k}
+    checkbounds(q, 1:(2k), 1:(2n))
+    checkbounds(p, 1:(2n), 1:(2k))
+    @inbounds for i in 1:k, j in 1:n
+        q[i, j] = p[j + n, i + k]
+    end
+    @inbounds for i in 1:k, j in 1:n
+        q[i, j + n] = -p[j, i + k]
+    end
+    @inbounds for i in 1:k, j in 1:n
+        q[i + k, j] = -p[j + n, i]
+    end
+    @inbounds for i in 1:k, j in 1:n
+        q[i + k, j + n] = p[j, i]
+    end
+    return q
+end
+
+@doc raw"""
+    make_metric_compatible!(::SymplecticStiefel, Y, ::EuclideanMetric, p, X)
+
+TODO: Complete.
```

Comment:
can you be more complete here – is it maybe the change<_metric function? Without the details I am not sure what this function does, but we should try to find a better name.

Reply:
I will expand on the documentation and make it more clear what we're doing here, but it's basically the 'change_representer' function, without the correct Tangent vector space range. I am a bit unsure as to what would be a better name, and I'm considering maybe taking this function out as it doesn't quite do what we would like from a 'change_representer'-function, and might therefore not be so useful.

Reply:
Ok, so we also do not use this function anywhere? Then it might really be an idea to remove it. I am still not yet sure what it actually does ;)

Reply:
He he yeah that's true. It's a leftover from when I was working on the Euclidean projection-gradient conversion method. 
I managed to find a working 'change_representer'-function on the symplectic manifold, with the convex combination of two 
'metric compatible' functions like this one. (c1 and c2 from the specialization report.) On symplectic Stiefel though, I've not managed to find another Metric compatibility function to attempt the same 'convex combination' trick with.

Reply:
I removed the `make_metric_compatible(::Symplectic, ...)` function, along with the relevant tests.